src/transformers/models/ict/modeling_ict.py

```diff
@@ -0,0 +1,908 @@
+    ICT_GUIDED_UP_SAMPLER_START_DOCSTRING,
+)
+class ICTModel(ICTPretrainedGuidedUpsampler):
+    config_class = ICTConfig
+
+    def __init__(self, config: ICTConfig):
+        super().__init__(config)
+
+        if not isinstance(config.transformer_config, ICTTransformerConfig):
+            raise ValueError(
+                "config.transformer_config is expected to be of type ICTTransformerConfig but is of type"
+                f" {type(config.transformer_config)}."
+            )
+
+        if not isinstance(config.guided_upsampler_config, ICTGuidedUpsamplerConfig):
+            raise ValueError(
+                "config.guided_upsampler_config is expected to be of type ICTGuidedUpsamplerConfig but is of type"
+                f" {type(config.guided_upsampler_config)}."
+            )
+
+        transformer_config = config.transformer_config
+        guided_upsampler_config = config.guided_upsampler_config
+        self.config = config
+        self.transformer = ICTTransformerModel(transformer_config)
+        self.guided_upsampler = ICTGuidedUpsampler(guided_upsampler_config)
+
+        # Initialize weights and apply final processing
+        # self.post_init()
+
+    @add_start_docstrings_to_model_forward(ICT_GUIDED_UP_SAMPLER_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=ImageSuperResolutionOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        pixel_values: Optional[torch.Tensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, BaseModelOutput]:
+        r"""
+        Returns:
+
+        Example:
+         ```python
+         >>> import torch
+         >>> import numpy as np
+         >>> from PIL import Image
+         >>> import requests
+
+         >>> from transformers import AutoImageProcessor, ICTModel
+
+         >>> processor = AutoImageProcessor.from_pretrained("sheonhan/ict-imagenet-32")
+         >>> model = ICTModel.from_pretrained("sheonhan/ict-imagenet-32")
+
+         >>> url = "TODO"
+         >>> image = Image.open(requests.get(url, stream=True).raw)
+         >>> # prepare image for the model
+         >>> inputs = processor(image, return_tensors="pt")
+
+         >>> # forward pass
+         >>> with torch.no_grad():
+         ...     outputs = model(**inputs)
+
+         >>> output = TODO
+         ```"""
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        if pixel_values is None:
+            raise ValueError("You have to specify pixel_values")
+
+        height, width = pixel_values.shape[2:]
+
+        outputs = self.tranformer(
+            pixel_values,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        self.guided_upsampler(outputs)
```

Comment:
For the upsampler to process the images, it probably needs to be loaded ([the original repo does it this way](https://github.com/raywzy/ICT/blob/main/Guided_Upsample/src/dataset_my.py)), but I wasn't sure how to go about doing this

Reply:
The guided upsampler takes the original masked image + the output embedding of the transformer module as input:
```
class ICTGuidedUpsampler:
    def   __init__(self, config...):
         ...
    def forward(self, pixel_values, hidden_states):
         ...
```
pixel_values is just the output of the `ICTImageProcessor `class, which resizes, normalizes and applies other preprocessing steps to the image. The content of the ICT Dataset class should go into the `ICTImageProcessor `class but we don't support and won't use any of the augmentation steps so it will be much more concise.